pkg/identity/ciprovider/principal.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciprovider
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"fmt"
+	"html/template"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/sigstore/fulcio/pkg/certificate"
+	"github.com/sigstore/fulcio/pkg/config"
+	"github.com/sigstore/fulcio/pkg/identity"
+	"gopkg.in/yaml.v3"
+)
+
+type Config struct {
+	Providers map[config.IssuerType]Provider
+}
+type Provider struct {
+	Subject     string
+	Extensions  certificate.Extensions
+	Uris        []string
```

Comment:
Is URIs the SubjectAlternativeName? If so, should we call it `SubjectAlternativeName`? And not make it repeated, since we only support a single SAN?